src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -848,6 +848,18 @@ public function compileInsert(Builder $query, array $values)
         return "insert into $table ($columns) values $parameters";
     }
 
+    /**
+     * Compile an insert statement, that ignore errors and silently skips erronous rows, 
+     * into SQL.
+     */
+    public function compileInsertOrIgnore(Builder $query, array $values)
+    {
+        $insert = $this->compileInsert($query, $values);
+
+        $insert = str_replace('insert', 'insert ignore', $insert);
```

Comment:
Just a passer by comment but won't this mistakenly replace column names with the word insert them? e.g. if I have a column called `was_inserted` this would replace it with `was_insert or ignoreed` wouldn't it?

Reply:
Thanks, thats seems like an immediate bug right there. I was reluctant to copy paste the insertion code, but forgot to control the replace. I see this needs cleaner control. 
Thank you for the comment.